### ISSUE TITLE
Images do not load with OkHttp

### ISSUE THREAD

**Issue by madamczewski** [bumptech/glide#404]
**Glide Version/Integration library (if any): 3.6.0-SNAPSHOT / okhttp-integration:1.2.2
Device/Android Version: Sony Xperia T / 4.1, Samsung S3 mini / 4.2, Nexus 5 / 5.1
Issue details/Repro steps:**

I receive following exception when I try to load images using OkHttp:
> java.io.IOException: Illegal character in path at index 87: http://jafolders.com/uploads/pages/55254ff073310028737b0000/normal_image_1428508658.jpg
com.bumptech.glide.load.model.Headers$1@41bc2110

Example code:
```java
Glide.get(this).register(GlideUrl.class, InputStream.class, new OkHttpUrlLoader.Factory(new OkHttpClient()));

Glide.with(this)
        .load(imageUrl)
        .into(new ViewTarget<ImageView, GlideDrawable>(imageView) {
                    @Override
                    public void onResourceReady(GlideDrawable resource, GlideAnimation<? super GlideDrawable> glideAnimation) {
                        imageView.setImageDrawable(resource);
                    }

                    @Override
                    public void onLoadFailed(Exception e, Drawable errorDrawable) {
                        Log.e(TAG, "error: "  + e.getMessage());
                    }
                }); 
```

**Comment by TWiStErRob** [comment]
Since the URL's length is 87 the index 87 after the end. I don't know your source for the url, but I guess there's a new line after the `g`, try trimming the `imageUrl` before giving it to `.load()`

**Comment by sjudd** [comment]
Can you also provide the full stack trace? Thanks!

**Comment by madamczewski** [comment]
@TWiStErRob I don't think it is a case of wrong url as everything works fine without okhttp, but I've tried to trim it and result is the same.

@sjudd Here is the full stack trace
http://pastebin.com/YfCS0NRL

You can find image url in exception if you'd like to test it :)

**Comment by sjudd** [comment]
It looks like the result of the headers object's toString method is just being appended to the url:

```
http://jafolders.com/uploads/pages/5511209e7331006798c6ee08/normal_image_1427185832.jpg
    com.bumptech.glide.load.model.Headers$1@41767f60
```

**Comment by TWiStErRob** [comment]
So it is the new line after all, but it is appended by [OkHttpStreamFetcher.loadData](https://github.com/bumptech/glide/blob/v3.5.2/integration/okhttp/src/main/java/com/bumptech/glide/integration/okhttp/OkHttpStreamFetcher.java#L29) which calls `url.toString()` instead of `url.toStringUrl()`. `LazyHeaders` has a `toString` and `Headers.NONE` should have one too.

It only happens on 3.6.0-SNAPSHOT it works with 3.5.2.

It's also called in `OkHttpStreamFetcher.getId`, I'm not sure it's correct there either, especially since it may include a `hashCode` and not the proper `toString` of the headers. `HttpUrlFetcher` and `VolleyStreamFetcher` on the other hand uses the correct getter and 
```
@Override public String getId() { glideUrl.getCacheKey(); }
```

**Comment by sjudd** [comment]
@TWiStErRob It looks like you're looking at the 3.5.2 version of the file? The most recent version uses toStringUrl: https://github.com/bumptech/glide/blob/master/integration/okhttp/src/main/java/com/bumptech/glide/integration/okhttp/OkHttpStreamFetcher.java.

I'll look in to this a bit more.

**Comment by sjudd** [comment]
I plugged the OkHttp integration from master into the flickr sample and everything works fine. I also tried replacing all loads with the url in questIon, which also worked fine: http://jafolders.com/uploads/pages/55254ff073310028737b0000/normal_image_1428508658.jpg.

It looks like you might be appending an object to your image url? Can you provide a little more code as to where you get the url?

**Comment by sjudd** [comment]
Ah sorry I should have noticed earlier: you need to depend on the 1.3.0-SNAPSHOT version of the okhttp integration library along with the 3.6.0-SNAPSHOT version of Glide. Doing so should fix these errors.

**Comment by madamczewski** [comment]
Yeah it works :) Thank you guys for quick response.
Btw I was looking for any snapshot of okhttp integration before, but could not find it. 
How can I know that there is a snapshot version available?

**Comment by TWiStErRob** [comment]
Follow #407 for snapshots.

**Comment by sjudd** [comment]
Closed #407 after adding a new Snapshots page, let us know if anything still isn't clear.
